chutney/environment/src/main/java/com/chutneytesting/environment/domain/EnvironmentService.java

```diff
@@ -42,9 +44,11 @@ public class EnvironmentService {
 
     private final Logger logger = LoggerFactory.getLogger(EnvironmentService.class);
     private final EnvironmentRepository environmentRepository;
+    private final List<UpdateEnvironmentHandler> updateEnvironmentHandlers;
 
-    public EnvironmentService(EnvironmentRepository environmentRepository) {
+    public EnvironmentService(EnvironmentRepository environmentRepository, List<UpdateEnvironmentHandler> updateEnvironmentHandlers) {
         this.environmentRepository = environmentRepository;
+        this.updateEnvironmentHandlers = updateEnvironmentHandlers;
```

Comment:
Could be better to have a non null `updateEnvironmnetHandlers`.
Test for null value (Optional) and set to emptyList.
Then no need for null check in methods.